website/source/docs/guides/semaphore.html.md

```diff
@@ -65,75 +67,106 @@ This is something like:
+This body is opaque to Consul but can be useful for human operators.
+
 The `<session>` value is the ID returned by the call to
 [`/v1/session/create`](/api/session.html#session_create).
 
-`body` can be used to associate a meaningful value with the contender. This is opaque
-to Consul but can be useful for human operators.
-
 The call will either return `true` or `false`. If `true`, the contender entry has been
 created. If `false`, the contender node was not created; it's likely that this indicates
 a session invalidation.
 
-The next step is to use a single key to coordinate which holders are currently
+The next step is to create a single key to coordinate which holders are currently
 reserving a slot. A good choice for this lock key is simply `<prefix>/.lock`. We will
 refer to this special coordinating key as `<lock>`.
 
-The current state of the semaphore is read by doing a `GET` on the entire `<prefix>`:
+This is done with:
 
 ```text
-curl http://localhost:8500/v1/kv/<prefix>?recurse
+curl -X PUT -d <body> http://localhost:8500/v1/kv/<lock>?cas=0
  ```
 
-Within the list of the entries, we should find the `<lock>`. That entry should hold
-both the slot limit and the current holders. A simple JSON body like the following works:
+Since the lock is being created, a `cas` index of 0 is used so that the key is only put if it does not exist.
+
+`body` should contain both the intended slot limit for the semaphore and the session ids
+of the current holders (initially only of the creator). A simple JSON body like the following works:
 
 ```text
 {
-    "Limit": 3,
+    "Limit": 2,
     "Holders": {
-        "4ca8e74b-6350-7587-addf-a18084928f3c": true,
-        "adf4238a-882b-9ddc-4a9d-5b6758e4159e": true
+        "<session>"
     }
 }
 ```
 
-When the `<lock>` is read, we can verify the remote `Limit` agrees with the local value. This
-is used to detect a potential conflict. The next step is to determine which of the current
-slot holders are still alive. As part of the results of the `GET`, we have all the contender
+The current state of the semaphore is read by doing a `GET` on the entire `<prefix>`:
+
+```text
+curl http://localhost:8500/v1/kv/<prefix>?recurse
+ ```
+
+Within the list of the entries, we should find two keys: the `<lock>` and the
+contender key ‘<prefix>/<session>’. 
+
+```text
+[
+  {
+    "LockIndex": 0,
+    "Key": "<lock>",
+    "Flags": 0,
+    "Value": "eyJMaW1pdCI6IDIsIkhvbGRlcnMiOnsiPHNlc3Npb24+In19",
+    "Session": "",
+    "CreateIndex": 898,
+    "ModifyIndex": 901
+  },
+  {
+    "LockIndex": 1,
+    "Key": "<prefix>/<session>",
+    "Flags": 0,
+    "Value": null,
+    "Session": "<session>",
+    "CreateIndex": 897,
+    "ModifyIndex": 897
+  }
+]
+```
+Note that the `Value` we embedded into `<lock>` is Base64 encoded when returned by the API.
+
+When the `<lock>` is read and its `Value` is decoded, we can verify the remote `Limit` agrees with the `Holders` count. 
```

Comment:
remote? not clear what that is referring to.

Reply:
true, going to remove that word